frontend/packages/topology/src/components/ElementWrapper.tsx

```diff
@@ -13,12 +13,9 @@ const ElementComponent: React.FC<ElementWrapperProps> = observer(({ element }) =
     () => element.getController().getComponent(element.getKind(), element.getType()),
     [element],
   );
-  if (!element.isVisible()) {
-    return null;
-  }
   if (isEdge(element)) {
-    const source = element.getSource();
-    const target = element.getTarget();
+    const source = element.getSourceAnchorNode();
+    const target = element.getTargetAnchorNode();
     if ((source && !source.isVisible()) || (target && !target.isVisible())) {
       return null;
     }
```

Comment:
Not sure why the `isVisible` checks were done in `ElementComponent` but it makes sense to move all these checks to `ElementWrapper` instead.

Also please change the comment of this component so that `interaction handlers` => `behaviors`